workload/terraform/modules/mgmtvm/main.tf

```diff
@@ -0,0 +1,144 @@
+# Resource block to generate a random string for the local admin password
+resource "random_string" "AVD_local_password" {
+  length           = 16
+  special          = true
+  min_special      = 2
+  override_special = "*!@#?"
+}
+
+# User Assigned Managed Identity for the MGMT VM to join storage accounts to the domain
+resource "azurerm_user_assigned_identity" "stguai" {
+  name                = "id-storage-${var.prefix}-${var.deployment_environment}-001"
```

Comment:
input variable with the name "deployment_environment" has not been declared

Reply:
addressed